### PR TITLE
Add stable/prometheus-operator values

### PR DESCRIPTION
**Pull request by jasonBirchall**
**Overview**
---
This PR connects to https://github.com/ministryofjustice/cloud-platform/issues/610. The overall idea of this change is to upgrade the chart and component versions of [prometheus-operator](https://github.com/coreos/prometheus-operator/tree/master/helm#this-chart-is-deprecated) from the depreciated in the CoreOS repository to the ported [version](https://github.com/helm/charts/blob/master/stable/prometheus-operator/values.yaml) (2.1.1) in [helm/stable](https://github.com/helm/charts/tree/master/stable/prometheus-operator#prometheus-operator). 

Component change:
```
Prometheus:   v2.6.0 -> v2.6.1
```

**Testing**
---
`cloud-platform-test-4.integration.dsd.io`
https://login.apps.cloud-platform-test-4.k8s.integration.dsd.io
https://prometheus.apps.cloud-platform-test-4.k8s.integration.dsd.io
https://alertmanager.apps.cloud-platform-test-4.k8s.integration.dsd.io
https://grafana.apps.cloud-platform-test-4.k8s.integration.dsd.io

```bash
> helm list
NAME               	REVISION	UPDATED                 	STATUS  	CHART                    	APP VERSION	NAMESPACE
prometheus-operator	1       	Tue Feb  5 17:55:30 2019	DEPLOYED	prometheus-operator-2.1.1	0.26.0     	monitoring
```

**What does this PR include?**
---
1. This PR includes a reference change. Anything referencing `kube-prometheus` or `kube_prometheus` has been changed accordingly. 
2. A removal of the `storageclass` creation and instead a grafana secret file containing all Grafana sensitive data. 
3. A removal of the old `helm_release.prometheus_operator`.
4. Addition of a new `helm_release.prometheus_operator`, replacing the core kube_prometheus.
5. A local_exec containing the relevant commands to remove any leftovers when performing a `terraform destroy`.
6. Removal of the file `kube-prometheus.yaml.tpl` as this is no longer required. 
7. Amendments to the old `prometheus-operator.yaml.tpl` file of which is the amalgamation of core components, previously in `kube-prometheus.yaml.tpl`.
8. Interpolation of the Grafana secret from encrypted `tfvars` file.

**Why?**
---
1. All pointers to `kube-prometheus` are now obsolete and will fail.
2. `Kops` creates a nice default `storageClass` so I decided to use this.
3.  The `helm_release.prometheus_operator` has been superseded and values in the file were now irrelevant.
4. All values have been amalgamated into a single values file.
5. When we destroy prometheus, all CRD's.
6. File no longer required. Please see 3.
7. Please see above. 
8. Kubernetes secret is created to follow best practice. This is then interpolated by Terraform.

**How to apply this in Live**
---
After merge to master.
```bash
> cd terraform/cloud-platform-components
> terraform init
> terraform workspace select cloud-platform-live-0
> terraform apply
```

### CODE REVIEW

File: terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl [hunk 1]
```diff
@@ -1,26 +1,1140 @@
...
+    enabled: true
+    namespace: kube-system
+  ## Create a servicemonitor for the operator
+  ##
+  serviceMonitor:
+    selfMonitor: true
+
+  ## Resource limits & requests
+  ##
+  resources: {}
```
[the remark this answers is not in the document]
Comment (jasonBirchall) [reply terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl]: Do you mean for the Prometheus-operator pod or generally across all components in this template like grafana etc?

File: terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl [hunk 2]
```diff
@@ -1,26 +1,1140 @@
...
+
+    ## Time duration Alertmanager shall retain data for. Default is '120h', and must match the regular expression
+    ## [0-9]+(ms|s|m|h) (milliseconds seconds minutes hours).
+    ##
+    retention: 120h
+
+    ## Storage is the definition of how storage will be used by the Alertmanager instances.
+    ## ref: https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/storage.md
+    ##
+    storage: {}
```
[the remark this answers is not in the document]
Comment (jasonBirchall) [reply terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl]: I have no problems with this. Will give it the same as Prometheus.

File: terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (alkar) [reply terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl]: I was thinking prometheus-operator but perhaps there's no point, I just looked at the current usage and it's really nothing, burstable should be fine.

File: terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl [hunk 3]
```diff
@@ -1,26 +1,1138 @@
...
+  ##
+  prometheusConfigReloaderImage:
+    repository: quay.io/coreos/prometheus-config-reloader
+    tag: v0.27.0
+
+  ## Hyperkube image to use when cleaning up
+  ##
+  hyperkubeImage:
+    repository: k8s.gcr.io/hyperkube
+    tag: v1.11.0
```
Comment (alkar) [inline terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl]: `v1.11.7` [is latest](https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/hyperkube?gcrImageListsize=30&gcrImageListquery=%255B%257B_22k_22_3A_22_22_2C_22t_22_3A10_2C_22v_22_3A_22_5C_221.11_5C_22_22%257D%255D&gcrImageListsort=-uploaded), can we please bump that as it no doubt includes a ton of fixes!

Comment (jasonBirchall) [reply terraform/cloud-platform-components/templates/prometheus-operator.yaml.tpl]: fixed in ` [25ae4aa](/ministryofjustice/cloud-platform-infrastructure/pull/148/commits/25ae4aabf15d16152b53ef3da4e09d3404eb72f0)`

**Pull request merged**